### PR TITLE
rename `permutation_invariant_training`

### LINKED ISSUE #729: Unify functional metrics names
## 🚀 Feature

let's unify the names of the metrics across the package and use `sklearn` and good [reference](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) 

### Motivation

let's apply the same assumption for all domains and preferably use fully expanded names (as a new person to the domain may not know the particular abbreviation)

### Pitch

be consistent across domains and close to standard

### Alternatives

each of these can be taken as a separate PR

- [ ] `f1` -> `f1_score`
- [ ] `fbeta` -> `fbeta_score`
- [ ] `hinge` -> `hinge_loss`
- [ ] `pit` -> `permutation_invariant_training`
- [ ] `pit_permutate` -> ???
- [ ] `psnr` -> `peak_signal_noise_ratio`
- [ ] `sdr` -> `signal_distortion_ratio` #711
- [ ] `si_sdr` -> `scale_invariant_signal_distortion_ratio` #711
- [ ] `snr` -> `signal_noise_ratio` #712
- [ ] `si_snr` -> `scale_invariant_signal_noise_ratio` #712
- [ ] `ssim` -> `structual_similarity_index_measure`
- [x] `wer` -> `word_error_rate` #714

### PR DESCRIPTION
**Pull request by puhuk**
To unify `pit` to `permutation_invariant_training`

## What does this PR do?

Part of #729, To unify `pit` to `permutation_invariant_training`

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by stancld — COMMENTED** [review]
Please, bear in mind we need to keep `PIT` and `pit` for backward compatibility. Please see #731 :] 

<hr>

- [ ] deprecation warnings & back compatibility
- [x] add changelog

File: torchmetrics/functional/audio/pit.py [hunk 1]
```diff
@@ -131,7 +213,8 @@ def pit(
         >>> # [batch, spk, time]
         >>> preds = torch.tensor([[[-0.0579,  0.3560, -0.9604], [-0.1719,  0.3205,  0.2951]]])
         >>> target = torch.tensor([[[ 1.0958, -0.1648,  0.5228], [-0.4100,  1.1942, -0.5103]]])
-        >>> best_metric, best_perm = pit(preds, target, scale_invariant_signal_distortion_ratio, 'max')
+        >>> best_metric, best_perm = pit(preds, target, \
+            scale_invariant_signal_distortion_ratio, 'max')
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py]:
```suggestion
        >>> best_metric, best_perm = pit(preds, target, scale_invariant_signal_distortion_ratio, 'max')
```

File: torchmetrics/functional/audio/pit.py [hunk 2]
```diff
@@ -101,13 +101,95 @@ def _find_best_perm_by_exhuastive_method(
...
+        best_metric of shape [batch],
+        best_perm of shape [batch]
+
+    Example:
+        >>> from torchmetrics.functional.audio import scale_invariant_signal_distortion_ratio
+        >>> # [batch, spk, time]
+        >>> preds = torch.tensor([[[-0.0579,  0.3560, -0.9604], [-0.1719,  0.3205,  0.2951]]])
+        >>> target = torch.tensor([[[ 1.0958, -0.1648,  0.5228], [-0.4100,  1.1942, -0.5103]]])
+        >>> best_metric, best_perm = permutation_invariant_training(preds, target, \
+            scale_invariant_signal_distortion_ratio, 'max')
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:136]:
```suggestion
        >>> best_metric, best_perm = permutation_invariant_training(
        ...     preds, target, scale_invariant_signal_distortion_ratio, 'max')
```

**Review by Borda — APPROVED** [review]

File: torchmetrics/functional/audio/pit.py [hunk 3]
```diff
@@ -178,29 +180,48 @@ def pit(
...
+    )
+    return permutation_invariant_training(preds, target, metric_func, eval_func, **kwargs)
+
+
+def pit_permutate(preds: Tensor, perm: Tensor) -> Tensor:
+    """permutate estimate according to perm.
+
+    Args:
+        preds (Tensor): the estimates you want to permutate, shape [batch, spk, ...]
+        perm (Tensor): the permutation returned from permutation_invariant_training, shape [batch, spk]
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:221]:
```suggestion
        preds: the estimates you want to permutate, shape [batch, spk, ...]
        perm: the permutation returned from permutation_invariant_training, shape [batch, spk]
```

File: torchmetrics/functional/audio/pit.py [hunk 4]
```diff
@@ -101,10 +101,11 @@ def _find_best_perm_by_exhuastive_method(
     return best_metric, best_perm  # shape [batch], shape [batch, spk]
 
 
-def pit(
+def permutation_invariant_training(
     preds: torch.Tensor, target: torch.Tensor, metric_func: Callable, eval_func: str = "max", **kwargs: Dict[str, Any]
 ) -> Tuple[Tensor, Tensor]:
-    """Permutation invariant training (PIT). The PIT implements the famous Permutation Invariant Training method.
+    """Permutation invariant training. The permutation_invariant_training implements the famous Permutation
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:107]:
```suggestion
    """Permutation invariant training (PIT). The ``permutation_invariant_training`` implements the famous Permutation
```

File: torchmetrics/functional/audio/pit.py [hunk 5]
```diff
@@ -178,29 +180,48 @@ def pit(
     return best_metric, best_perm
 
 
-def pit_permutate(preds: Tensor, perm: Tensor) -> Tensor:
-    """permutate estimate according to perm.
+def pit(
+    preds: torch.Tensor, target: torch.Tensor, metric_func: Callable, eval_func: str = "max", **kwargs: Dict[str, Any]
+) -> Tuple[Tensor, Tensor]:
+    """Permutation invariant training. The pit implements the famous Permutation Invariant Training method.
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:186]:
```suggestion
    """Permutation invariant training. The ``pit`` implements the famous Permutation Invariant Training method.
```

**Comment by Borda** [comment]
@stancld fixed now :rabbit:

**Review by stancld — APPROVED** [review]

**Pull request merged**